filterpage.html

```diff
@@ -0,0 +1,396 @@
+    </div>
+
+    <script>
+      let selectedMakeFilter = [];
+      let selectedBodyFilter = [];
+      let openFilterButton = document.querySelector(".open-filter__btn");
+      let selection = document.querySelector(".selection-wrapper");
+
+      //onclick of the open filter button the firstpage hides and the next filter selection page opens
+      let openFirstPageWrapper = document.querySelector(
+        ".open-filter-page-wrapper"
+      );
+      openFilterButton.addEventListener("click", function () {
+        selection.classList.remove("hide");
+        openFirstPageWrapper.classList.add("hide");
+      });
+
+      //onclick of the cross symbol the first page opens and the filter selection page close
+      let crossButton = document.querySelector(".cross-symbol-top");
+      crossButton.addEventListener("click", function () {
+        selection.classList.add("hide");
+        openFirstPageWrapper.classList.remove("hide");
+      });
+
+      //onclick of the arrow the Make list open and collapse
+      let upwardArrow = document.querySelector(".arrow");
+      let makeListWrapper = document.querySelector(".make-list-wrapper");
+      upwardArrow.addEventListener("click", function (e) {
+        makeListWrapper.classList.toggle("collapse");
+        e.target.classList.toggle("downward");
+      });
+
+      //here we are checking the selected type is checkbox and we have checked the checkbox then we push it into array
+      //if we unselected the checkbox using splice the selected index is removed
+      makeListWrapper.addEventListener("change", function (event) {
+        if (event.target.matches('input[type="checkbox"]')) {
+          const checkbox = event.target;
+          if (checkbox.checked) {
+            selectedMakeFilter.push(checkbox.value);
+          } else {
+            const index = selectedMakeFilter.indexOf(checkbox.value);
+            if (index > -1) {
+              selectedMakeFilter.splice(index, 1);
+            }
+          }
+        }
+      });
+
+      //onclick of the apply filter the selection page hides and first page open and in result on first page we can see the selected status
+      let applyFilter = document.querySelector(".apply-filters__btn");
+      let result = document.querySelector(".selection-status");
+      applyFilter.addEventListener("click", function () {
+        selection.classList.add("hide");
+        openFirstPageWrapper.classList.remove("hide");
+        result.innerHTML =
+          selectedMakeFilter.length > 0
+            ? `<div>Make: <span class="selected-result">${selectedMakeFilter.join(
+                ", "
+              )} </span></div>`
+            : "Make: No selected filters";
+      });
+
+      //onclick of the arrow the Body list open and collapse
+      let bodyTypeArrow = document.querySelector(".body-arrow");
+      let bodyListWrapper = document.querySelector(".body-list-wrapper");
+      bodyTypeArrow.addEventListener("click", function (e) {
```

Comment:
Instead of adding eventListener to each collapsible with a different className can we use the same className?
If we want to add 3rd filter section, then according to the current implementation we need to add another className for that filter and then need to add eventListner as well.

Reply:
but how does it know user click on which part? let me figure it out!